src/components/UserInfo/UserInfo.tsx

```diff
@@ -1,5 +1,24 @@
-import React from 'react';
+import { User } from '../../types/user';
+import { Comment } from '../../types/comment';
 
-export const UserInfo: React.FC = () => (
-  <>Put the user here</>
-);
+type Props = {
+  user: User | Comment,
+  dataType: 'post' | 'comment',
+};
+
+export const UserInfo: React.FC<Props> = ({ user, dataType }) => {
```

Comment:
User info and post info - different things. Would be good to separate your logic to separate components.
In practice, better to split it into small components rather than have one but with complicated logic.

If you need more explanations - feel free to ping me in the slack and discuss this point